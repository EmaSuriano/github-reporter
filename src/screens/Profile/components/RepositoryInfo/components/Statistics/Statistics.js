import React from "react";
import { arrayOf, number, shape, string } from "prop-types";
import { isEmpty } from "ramda";
import { Doughnut } from "react-chartjs-2";
import { Box, Text } from "grommet";
import styled from "styled-components";

import NoDataFound from "../NoDataFound";

const StatisticsBox = styled(Box)`
  background-color: #ffffff;
  border: 1px solid #eee9e9;
  border-radius: 0.5rem;
`;

const Statistics = ({ configuration, data, id, title }) => (
  <StatisticsBox justify="center" align="center" gap="medium">
    <Text>{title}</Text>
    {isEmpty(data.labels) ? (
      <NoDataFound />
    ) : (
      <Doughnut id={id} data={data} {...configuration} />
    )}
  </StatisticsBox>
);

Statistics.defaultProps = {
  title: ""
};

Statistics.propTypes = {
  data: shape({
    labels: arrayOf(string),
    datasets: arrayOf(
      shape({
        backgroundColor: arrayOf(string),
        data: arrayOf(number)
      })
    )
  }),
  id: string,
  title: string
};

export default Statistics;
