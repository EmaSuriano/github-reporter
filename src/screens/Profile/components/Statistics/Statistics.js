import React from "react";
import { arrayOf, number, shape, string } from "prop-types";
import { Doughnut } from "react-chartjs-2";

const Statistics = ({ data, configuration, title }) => (
  <div className="profile--box">
    <h1>{title}</h1>
    <Doughnut data={data} {...configuration} />
  </div>
);

Statistics.defaultProps = {
  title: ""
};

Statistics.propTypes = {
  data: shape({
    labels: arrayOf(string),
    datasets: arrayOf(
      shape({
        backgroundColor: arrayOf(string),
        data: arrayOf(number),
        hoverBackgroundColor: arrayOf(string)
      })
    )
  })
};

export default Statistics;
