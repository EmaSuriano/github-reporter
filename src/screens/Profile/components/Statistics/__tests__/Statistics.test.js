import React from "react";
import { shallow } from "enzyme";

import Statistics from "../Statistics";

import data from "./fixtures/data/data";
import configuration from "./fixtures/configuration/configuration";

const propsWithLabels = {
  data,
  configuration,
  title: "Title"
};

const propsWithoutLabels = {
  data: {},
  title: "Title"
};

describe("Statistics Component", () => {
  let wrapper;

  describe("render", () => {
    describe("when labels are empty", () => {
      beforeEach(() => {
        wrapper = shallow(<Statistics {...propsWithoutLabels} />);
      });

      it("should match with the html structure of snapshot", () =>
        expect(wrapper).toMatchSnapshot());
    });

    describe("when have labels", () => {
      beforeEach(() => {
        wrapper = shallow(<Statistics {...propsWithLabels} />);
      });

      it("should match with the html structure of snapshot", () =>
        expect(wrapper).toMatchSnapshot());
    });
  });
});
