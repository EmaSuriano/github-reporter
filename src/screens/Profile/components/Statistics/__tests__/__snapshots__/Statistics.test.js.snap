// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics Component render when have labels should match with the html structure of snapshot 1`] = `
<div
  className="statistic--box"
>
  <Doughnut
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "#DS25F1",
              "#21FAAC",
            ],
            "data": Array [
              100,
              32,
              46,
            ],
          },
        ],
        "labels": Array [
          "JavaScript",
          "PHP",
          "HTML",
        ],
      }
    }
    height={300}
    options={
      Object {
        "legend": Object {
          "labels": Object {
            "boxWidth": 12,
            "fontSize": 12,
            "padding": 15,
            "usePointStyle": true,
          },
          "position": "left",
        },
        "maintainAspectRatio": false,
        "responsive": false,
        "tooltips": Object {
          "enabled": true,
        },
      }
    }
    width={400}
  />
</div>
`;

exports[`Statistics Component render when labels are empty should match with the html structure of snapshot 1`] = `
<div
  className="statistic--box"
>
  <Doughnut
    data={Object {}}
  />
</div>
`;
